A32NX/html_ui/Pages/A32NX_Core/A32NX_TransitionAltitude.js

```diff
@@ -0,0 +1,345 @@
+class A32NX_TransitionAltitude {
+    init() {
+        console.log('A32NX_TransitionAltitude init');
+        this.currentDeparture = "";
+        this.currentArrival = "";
+        this.offline = false;
+        this.checkStart = true;
+        this.checkStartSec = 0;
+        this.testAirport = "";
+        this.trySendRequest();
+    }
+
+    update(_deltaTime, _core) {
+        if (this.offline === true) {
+            this.offlineTransAlt();
+        }
+        this.tryCheckAPI(_deltaTime);
+    }
+
+    async trySendRequest() {
+        await NXApi.getAirport("KLAX")
+            .then((data) => {
+                this.testAirport = data.transAlt;
+            });
+    }
+
+    tryCheckAPI(_deltaTime) {
+        if (this.offline === false && this.checkStart === true) {
+            this.checkStartSec += _deltaTime;
+        }
+        if ((this.checkStartSec >= 60 * 1000) && (this.testAirport === "")) {
+            this.offline = true;
+            this.checkStart = false;
+        } else if ((this.checkStartSec >= 60 * 1000) && (this.testAirport !== "")) {
+            this.offline = false;
+            this.checkStart = false;
+            this.checkStartSec = 0;
+        }
+    }
```

Comment:
I'm guessing the code above is some kind of API timeout mechanism. I have a few questions:

- 60 seconds seems a bit long. For example, might this cause a user filled in transition altitude to be overriden after 30 seconds?
- This seems a concern that shouldn't be at this level of abstraction. It would make sense for `NXApi` to provide the timeout abstraction and perhaps even for it to contain a default transition altitude fallback list. This will improve readability of code.

Reply:
That need to decision. 30sec is short if players computer is slow.
I notice that in even MSFS loading screen, sometimes instrument already loaded and working start.

Reply:
If it needs a decision, then this isn't resolved yet.

Reply:
I changed that to hasTelex~~ in API. anyway telex and api are same server isn't it?